docs/src/Modal.doc.js

```diff
@@ -23,85 +32,497 @@ card(
+  const HEADER_ZINDEX = new FixedZIndex(10);
+  const modalZIndex = new CompositeZIndex([HEADER_ZINDEX]);
+
+  return (
+    <React.Fragment>
+      <Button
+        inline
+        text="View Modal"
+        onClick={() => setShouldShow(true)}
+      />
+      {shouldShow && (
+        <Layer zIndex={modalZIndex}>
+          <ModalWithHeading onDismiss={() => setShouldShow(false)} />
+        </Layer>
+      )}
+    </React.Fragment>
+  );
+}
+`}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Sub-heading"
+      description={`The \`subHeading\` is a container that can be used for subtext that provides additional context for the Modal. The sub-heading locks to the top under the heading.`}
+    >
+      <MainSection.Card
+        cardSize="lg"
+        iframeContent={`
+          <iframe src="https://codesandbox.io/embed/cool-hill-ngdjx?fontsize=14&hidenavigation=1&theme=light&view=preview"
+          style={{width: '100%', height:'500px', border:'0', borderRadius: '4px', overflow:'hidden'}}
+          title="Modal with Sub Heading"
+          allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
+          sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
+        ></iframe>
+        `}
+        defaultCode={`
+function SubHeadingExample(props) {
+  const ModalWithSubHeading = ({
+    onDismiss,
+  }) => {
+
+    return (
+      <Modal
+        accessibilityModalLabel="Resume account creation"
+        heading="Resume your work?"
+        subHeading="Welcome back to the business account creation process!"
+        align="left"
+        onDismiss={onDismiss}
+        footer={
+          <Flex alignItems="center" justifyContent="end" gap={2}>
+            <Button text="Cancel"/>
+            <Button inline color="red" text="Resume"/>
+          </Flex>
+        }
+        size="sm"
+      >
+        <Box paddingX={8}>
+          <Text>Want to continue where you left off? Click "Resume" to continue creating your account or "Cancel" to start over.</Text>
+        </Box>
+      </Modal>
+    );
+  };
+
+  const [shouldShow, setShouldShow] = React.useState(true);
+  const HEADER_ZINDEX = new FixedZIndex(10);
+  const modalZIndex = new CompositeZIndex([HEADER_ZINDEX]);
+
+  return (
+    <React.Fragment>
+      <Button
+        inline
+        text="View Modal"
+        onClick={() => setShouldShow(true)}
+      />
+      {shouldShow && (
+        <Layer zIndex={modalZIndex}>
+          <ModalWithSubHeading onDismiss={() => setShouldShow(false)} />
+        </Layer>
+      )}
+    </React.Fragment>
+  );
+}
+`}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Sizes"
+      description={`Modal has 4 size options: small (\`sm\`), medium (\`md\`), large (\`lg\`) or a number representing a custom width.
```

Comment:
We really have three size options plus an escape hatch, right? Assuming we'd prefer people to stick with our t-shirt sizes, we should probably present those as "The Sizes", then mention that custom sizes are possible but not recommended.